docs/source/run_model_in_simulation.rst

```diff
@@ -28,31 +33,53 @@ simulator. Thus, in SMAUG, in order to run a model in gem5-Aladdin, we need to
 generate a dynamic trace for the kernels to be simulated as hardware blocks.
 
 First, we need to build the SMAUG tracer, which is an instrumented binary that
-will be executed to generate the dynamic trace::
+will be executed to generate the dynamic trace.
+
+.. code-block:: console
 
     make tracer -j4
 
 After running the above command, we will get a binary under :code:`build/bin/`
 named :code:`smaug-instrumented`. Then we will run the model created in
-:ref:`label_build_python_model` using the following command::
+:ref:`label_build_python_model` using the following command:
+
+.. code-block:: console
 
     build/bin/smaug-instrumented my_model_topo.pbtxt my_model_params.pb
 
-This generates a trace file named :code:`dynamic_trace_acc0.gz`.
+While the tracer is producing the dynamic trace, we will see outputs such as the
+following pair for each invocation of the hardware blocks::
+
+   dynamic_trace_acc0.gz: Starting to log at inst = 0.
+   dynamic_trace_acc0.gz: Stopping logging at inst 18199217.
+
+A trace file named :code:`dynamic_trace_acc0.gz` is generated.
+
+Note that for large models, this step can take a long time and the resulting
+trace file can be quite large. Sampling should be used to solve this, which is
+discussed in :ref:`label_apply_sampling`.
 
 Create gem5-Aladdin configuraion files
 ---------------------------------------
 
 gem5-Aladdin also requires two configuration files for running simulations. One
 file specifies SoC-level parameters such as each accelerator's ID, cache or DMA
 configurations, location of the corresponding dynamic trace file, and etc.  An
-example can found be in :code:`experiments/sims/smv/minerva/gem5.cfg`. The
-other file gives the parameters to be used for implementing the accelerator,
-such as parameters for applying loop unrolling and scratchpad partition. An
-example configuration file of implementing the hardware blocks in our `SMV`
-backend can be found in :code:`experiments/sims/smv/smv-accel.cfg`. We defer to
-the `gem5-Aladdin GitHub repo <https://github.com/harvard-acc/gem5-aladdin>`_
-and `gem5-Aladdin tutorial
+example can found be in :code:`experiments/sims/smv/minerva/gem5.cfg` (make
+sure the submodules are initialized by running :code:`git submodule update
+--init --recursive`). The other file gives the parameters to be used for
+implementing the accelerator, such as parameters for applying loop unrolling
+and scratchpad partition. An example configuration file of implementing the
+hardware blocks in our `SMV` backend can be found in
+:code:`experiments/sims/smv/smv-accel.cfg`.  If you don't make any changes to
+the `SMV` backend, both of these configuration files are sufficient to simualte
+any model run on `SMV`. You can change the parameters but it may lead to
+unexpected accelerator performance characteristics especially the ones in
```

Comment:
...accelerator performance characteristics, especially if you change :code:`smv-accel.cfg`.

Reply:
Done.